chromium_src/components/sync_device_info/device_info_sync_bridge.cc

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "../../../../components/sync_device_info/device_info_sync_bridge.cc"
+
+#include "base/threading/sequenced_task_runner_handle.h"
+
+namespace syncer {
+
+void DeviceInfoSyncBridge::DeleteDeviceInfo(const std::string& client_id,
+                                            base::OnceClosure callback) {
+  std::unique_ptr<WriteBatch> batch = store_->CreateWriteBatch();
+  change_processor()->Delete(client_id, batch->GetMetadataChangeList());
+  DeleteSpecifics(client_id, batch.get());
+  CommitAndNotify(std::move(batch), /*should_notify=*/true);
+  base::SequencedTaskRunnerHandle::Get()->PostDelayedTask(
+      FROM_HERE,
+      base::BindOnce(&DeviceInfoSyncBridge::OnDeviceInfoDeleted,
+                     weak_ptr_factory_.GetWeakPtr(), client_id,
+                     std::move(callback)),
+      base::TimeDelta::FromSeconds(1));
+}
+
+void DeviceInfoSyncBridge::OnDeviceInfoDeleted(const std::string& client_id,
+                                               base::OnceClosure callback) {
+  // Make sure the deleted device info is sent
+  if (change_processor()->IsEntityUnsynced(local_cache_guid_)) {
```

Comment:
We can't apply remote update of a client tag based self deleted entity because
In `ClientTagBasedRemoteUpdateHandler::ProcessUpdate`
```cpp
  // Handle corner cases first.
  if (entity == nullptr && data.is_deleted()) {
    // Local entity doesn't exist and update is tombstone.
    SyncRecordModelTypeUpdateDropReason(
        UpdateDropReason::kTombstoneForNonexistentInIncrementalUpdate, type_);
    DLOG(WARNING) << "Received remote delete for a non-existing item."
                  << " client_tag_hash: " << client_tag_hash << " for "
                  << ModelTypeToString(type_);
    return nullptr;
  }
```
So it will never reaches `DeviceInfoSyncBridge::ApplySyncChanges` and that is why we have to wait here